tests/server/test_server.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
```

Comment:
why are we creating a new `test_server.py` just for hot_config_reload functionality?

Reply:
this one is more of an integration test that actually spins up a server process via `start_server`. probably could even move this to test the cli tbh. felt different 🤷‍♂️

Reply:
Move to test_hot_reloading.py